examples/next-openai-pages/pages/chat/stream-chat-files/index.tsx

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable @next/next/no-img-element */
+import { useChat } from 'ai/react';
+import { useRef, useState } from 'react';
+
+export default function Page() {
+  const { messages, input, handleSubmit, handleInputChange, isLoading } =
+    useChat({
+      api: '/api/stream-chat',
+    });
+
+  const [files, setFiles] = useState<FileList | undefined>(undefined);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex flex-col p-2 gap-2">
+        {messages.map(message => (
+          <div key={message.id} className="flex flex-row gap-2">
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+
+            <div className="flex flex-col gap-2">
+              {message.content}
+
+              <div className="flex flex-row gap-2">
+                {message.files?.map((file, index) =>
+                  file.type === 'data-url' ? (
+                    file.dataUrl.includes('image/') ? (
+                      <img
+                        key={`${message.id}-${index}`}
+                        className="w-24 rounded-md"
+                        src={file.dataUrl}
+                        alt="image"
+                      />
+                    ) : file.dataUrl.includes('text/') ? (
+                      <div className="w-32 h-24 rounded-md text-xs ellipsis overflow-hidden p-2 text-zinc-500 border">
+                        {atob(file.dataUrl.split(',')[1])}
+                      </div>
+                    ) : null
+                  ) : null,
```

Comment:
could be collapsed if we had a single type

Reply:
done!